src/index.ts

```diff
@@ -201,6 +222,16 @@ function persistedOperationFromPayload(
     const hashFromPayload = options.hashFromPayload || defaultHashFromPayload;
     const hash = hashFromPayload(payload);
     if (typeof hash !== "string") {
+      const allowUnpersistedOperations =
+        options.allowUnpersistedOperation || (() => false);
+
+      if (allowUnpersistedOperations(payload)) {
+        const query = queryFromPayload(payload);
+        if (typeof query === "string") {
+          return query;
+        }
```

Comment:
Let's be more direct; we're ultimately doing `params.query = params.query` after all.

```suggestion
        if (typeof payload.query === "string") {
          return payload.query;
        }
```